xrmocap/human_perception/keypoints_estimation/mediapipe_estimator.py

```diff
@@ -0,0 +1,246 @@
+import cv2
+import logging
+import mediapipe as mp
+import numpy as np
+from tqdm import tqdm
+from typing import Tuple, Union
+from xrprimer.utils.ffmpeg_utils import video_to_array
+from xrprimer.utils.log_utils import get_logger
+
+from xrmocap.transform.convention.keypoints_convention import get_keypoint_num
+from .mmpose_top_down_estimator import MMposeTopDownEstimator
+
+
+class MediapipeEstimator(MMposeTopDownEstimator):
+
+    def __init__(self,
+                 mediapipe_kwargs: dict,
+                 bbox_thr: float = 0.0,
+                 logger: Union[None, str, logging.Logger] = None) -> None:
+        """Init a detector from mediapipe.
+
+        Args:
+            mediapipe_kwargs (dict):
+                A dict contains args of mediapipe.
+                refer to https://google.github.io/mediapipe/solutions/pose.html
+                in detail.
+            bbox_thr (float, optional):
+                Threshold of a bbox. Those have lower scores will be ignored.
+                Defaults to 0.0.
+            logger (Union[None, str, logging.Logger], optional):
+                Logger for logging. If None, root logger will be selected.
+                Defaults to None.
+        """
+        # build the pose model
+        mp_pose = mp.solutions.pose
+        self.pose_model = mp_pose.Pose(**mediapipe_kwargs)
+        self.bbox_thr = bbox_thr
+        self.logger = get_logger(logger)
+        self.convention = 'mediapipe_body'
+
+    def get_keypoints_convention_name(self) -> str:
+        """Get data_source from dataset type of the pose model.
+
+        Returns:
+            str:
+                Name of the keypoints convention. Must be
+                a key of KEYPOINTS_FACTORY.
+        """
+        return self.convention
+
+    def infer_single_img(self, img_arr: np.ndarray, bbox_list: list):
+        multi_kps2d = []
+        for bbox_dict in bbox_list:
```

Comment:
To align with `def infer_array()`, bbox_list shall be a nested list in shape `[n_person, 5]`, no dict inside.